### ISSUE TITLE
Install fails for release branch

### ISSUE THREAD

**Issue by ericbarefoot** [sededu/sededu#58]
Installing `sededu` from the release branch results in an incomplete install on Mac OSX, with several submodules omitted. Running `git submodule update --init --recursive` does not help, since one submodule seems to throw a fatal error:
```
error: Server does not allow request for unadvertised object d083b3c6a3237d0c0380fd8b2a094a46788dd5cd
Fetched in submodule path 'sededu/modules/rivers/rain-table_jeffskwang', but it did not contain d083b3c6a3237d0c0380fd8b2a094a46788dd5cd. Direct fetching of that commit failed.
```
As a result, I think this keeps all the other submodules from loading. 

Interestingly, if I clone the entire repository, I get a functional install with only the develop branch. I'm not sure what the fix is here, but it looks like there's something wrong with connecting Jeffrey's module to the release branch, aborting the cloning.

**Comment by amoodie** [comment]
@ericbarefoot  can you try the install again? Whatever method you prefer:
```
pip3 install pyqt sededu
```
or by cloning the repo. Thanks, LMK.

**Comment by ericbarefoot** [comment]
I get this error:

```
19:58:59 ericfoot ~ $  pip3 install pyqt sededu
Collecting pyqt
  Could not find a version that satisfies the requirement pyqt (from versions: )
No matching distribution found for pyqt
```

did you mean `pip3 install PyQT5-sip` ? because when I do `pip3 list` I get: 

```
19:59:05 ericfoot ~ $  pip3 list
Package             Version    
------------------- -----------
cycler              0.10.0     
GDAL                2.3.1      
kiwisolver          1.0.1      
matplotlib          3.0.0      
nose                1.3.7      
numpy               1.15.1     
pip                 18.0       
pyparsing           2.2.1      
PyQt5-sip           4.19.13    
python-dateutil     2.7.3      
rivers2stratigraphy 0.2.6      
scipy               1.1.0      
setuptools          40.2.0     
Shapely             1.6.4.post2
six                 1.11.0     
wheel               0.31.1  
```

**Comment by amoodie** [comment]
maybe
```pip3 install pyqt5```

or try without the `pyqt` at all, and just the `sededu` see what error you get on import? Thanks for your help.

**Comment by ericbarefoot** [comment]
OK, that got us one step closer. Now install runs smoothly and the whole thing starts up when I run the script. I can explore the GUI, but it crashes when I try to run any of your activities.

I'm gonna attach a text file here that includes the error text I got in the python session before it crashed, and also some sort of OSX error log that popped up.

[sededu_error_logs.txt](https://github.com/sededu/sededu/files/2611611/sededu_error_logs.txt)

**Comment by ericbarefoot** [comment]
Yes. I succeeded with `pip3 install pyqt5 sededu`, and I see what you're trying to do I think, if I read that utility right it passes "xdg-open _____" to the system? In that case it should be 
```
def open_file(filename):
    # utility for opening files
    platType = platform.system()
    if os.path.isfile(filename):
        if platType in {'Linux'}:
            subprocess.Popen(['xdg-open', filename])
        elif platType in {'Windows', 'Darwin'}:
            subprocess.Popen(['open', filename])
        else:
            print('unknown platform type')
    else:
        msg = NoFileMessageBox(filename)
        msg.exec_()
```
Because OSX opens things from the terminal with `open`, not `xdg-open`.

I'm at school now, but I'll check it out when I get home.

**Comment by ericbarefoot** [comment]
Did it! Opening things on OSX is tricky, and it might be something worthwhile for me to fool around with. For example, [this](https://github.com/sededu/stratigraphic_filter/blob/master/doc/theory.md) activities for stratigraphic_filter is a .md file, which OSX chooses to open according to the users preference, but if they haven't opened one before it might open in something ridiculous like XCode or powerpoint.

**Comment by amoodie** [comment]
Thanks, closed with #78
